configserver/src/main/java/com/yahoo/vespa/config/server/application/ApplicationCuratorDatabase.java

```diff
@@ -74,34 +74,56 @@ public boolean exists(ApplicationId id) {
     /**
      * Creates a node for the given application, marking its existence.
      */
-    public void createApplication(ApplicationId id, boolean writeAsJson) {
+    public void createApplication(ApplicationId id) {
         if ( ! id.tenant().equals(tenant))
             throw new IllegalArgumentException("Cannot write application id '" + id + "' for tenant '" + tenant + "'");
 
         try (Lock lock = lock(id)) {
             if (curator.exists(applicationPath(id))) return;
 
-            if (writeAsJson) {
-                var applicationData = new ApplicationData(id, OptionalLong.empty(), OptionalLong.empty());
-                curator.set(applicationPath(id), applicationData.toJson());
-            } else {
-                curator.create(applicationPath(id));
-            }
+            var applicationData = new ApplicationData(id, OptionalLong.empty(), OptionalLong.empty());
+            curator.set(applicationPath(id), applicationData.toJson());
+            modifyReindexing(id, ApplicationReindexing.empty(), UnaryOperator.identity());
+        }
+    }
+
+    /**
+     * Creates a node for the given application, marking its existence.
+     */
+    // TODO: Remove in Vespa 9
+    public void createApplicationInOldFormat(ApplicationId id) {
```

Comment:
Why cannot this method be removed now?  The only reference to this method is in the test below.

Reply:
There are config server out there (self-hosted) that have data in old format and need to be able to read it when upgrading